pkg/controller/argocd/policyrule.go

```diff
@@ -92,6 +92,20 @@ func policyRuleForRedisHa() []v1.PolicyRule {
 				"get",
 			},
 		},
+		{
```

Comment:
Can these be added by reconciler hook?

Reply:
It can't easily be added by the reconciler hook.  The hook expects an instance of the ArgoCD CR as a parameter, which we don't have in this function.  Even if we passed `nil` into the hook for the ArgoCD instance, there is no way to tell what the PolicyRule is for.  We would have to blindly assume that its for Redis HA.  This doesn't seem very clean.  I think to do it cleanly, we'd need another kind of hook into which we can pass additional information to identify what the passed object is for.  I don't think it's worth it.

For the corresponding change in ArgoCD itself, these values were added directly:
https://github.com/argoproj/argo-cd/pull/4800
https://github.com/argoproj/argo-cd/pull/5078

Reply:
Thanks for looking into it.  I was thinking since we have created the reconciler hook to isolate OpenShift specific code, we probably should use it to put all OpenShift roles in it.  I think we can pass ArgoCD CR to policyRuleForRedisHa() and we can add another argument to the hook func to idenifty the intent of the resource.  I agree that the code can get more complicated.  WDYT?

BTW, the reference to the corresponding change in ArgoCD is not quite the same situation as in here.  The OpenShift roles are added to an overlay specific to OpenShift.  We could generate an install yaml without OpenShift roles but it isn't needed.